### PR TITLE
replace std::tolower with Azure::Core::ToLower

### PR DESCRIPTION
**Pull request by Jinming-Hu**

### CODE REVIEW

**Review by vhvb1989 — APPROVED** [review]
Great, tks

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp [hunk 1]
```diff
@@ -3473,11 +3474,7 @@ namespace Azure { namespace Storage { namespace Blobs {
         std::set<std::string> metadataKeys;
         for (const auto& pair : options.Metadata)
         {
-          std::string key = pair.first;
-          std::transform(key.begin(), key.end(), key.begin(), [](unsigned char c) {
-            return static_cast<char>(std::tolower(c));
-          });
-          if (metadataKeys.insert(key).second == false)
+          if (metadataKeys.insert(Azure::Core::Details::ToLower(pair.first)).second == false)
```
Comment (RickWinter) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:3477]: Should not be using Details namespace outside of the assembly.  We need to move our ToLower function if you are going to take a dependency on it.

**Review by RickWinter — CHANGES_REQUESTED** [review]

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp (hunk 1 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:3477]: `ToLower` is a function provided in Core sdk. So this is a change core team should do, not storage team.

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:3477]: Or are you suggesting that we should just keep things as it is. Should I just close this PR?

Comment (ahsonkhan) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:3477]: FYI, @JeffreyRichter

**Pull request closed without merging**